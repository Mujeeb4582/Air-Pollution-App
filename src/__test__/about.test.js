import React from 'react';
import { render, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import About from '../pages/about';

const mockStore = configureMockStore();
const store = mockStore({
  airPollution: {
    loading: false,
    error: null,
    pollutionData: {
      countryName: 'United States',
      countrySymbol: 'US',
      list: [
        {
          main: { aqi: 100 },
        },
        {
          dt: 1609459200,
          components: {
            pm25: 12,
            pm10: 20,
            no2: 30,
            so2: 40,
            co: 50,
            o3: 60,
          },
        },
      ],
    },
  },
});

test('renders the pollution data table', async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <About />
    </Provider>,
  );
  const table = await waitFor(() => getByTestId('pollution-table'));
  expect(table).toBeInTheDocument();
});

test('renders About page correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <About />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
