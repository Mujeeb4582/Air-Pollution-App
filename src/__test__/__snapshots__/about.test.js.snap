// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  correctly 1`] = `
<div
  className="text-center"
  data-testid="pollution-table"
>
  <svg
    className="rsm-svg "
    style={
      Object {
        "backgroundColor": "#da4c7a",
        "fill": "#2a354d",
      }
    }
    viewBox="0 0 800 600"
  >
    <g
      className="rsm-geographies "
      data-testid="map"
    />
  </svg>
  <div
    style={
      Object {
        "margin": "16px 0",
        "textAlign": "center",
      }
    }
  >
    <h4>
      United States
      , 
      US
    </h4>
    <h6>
      Date: 
      1/19/1970
        
      <span>
        Air Quality Index: 
      </span>
      100
    </h6>
  </div>
  <table
    style={
      Object {
        "backgroundColor": "#da4c7a",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <thead>
      <tr
        className="fs-4"
      >
        <th
          className="py-2"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          Gases
        </th>
        <th
          className="py-2"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          Pollutant concentration in μg/m3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          PM25
        </td>
        <td
          className="py-3 fs-5"
        >
          12
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          PM10
        </td>
        <td
          className="py-3 fs-5"
        >
          20
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          NO2
        </td>
        <td
          className="py-3 fs-5"
        >
          30
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          SO2
        </td>
        <td
          className="py-3 fs-5"
        >
          40
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          CO
        </td>
        <td
          className="py-3 fs-5"
        >
          50
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          O3
        </td>
        <td
          className="py-3 fs-5"
        >
          60
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders About page correctly 1`] = `
<div
  className="text-center"
  data-testid="pollution-table"
>
  <svg
    className="rsm-svg "
    style={
      Object {
        "backgroundColor": "#da4c7a",
        "fill": "#2a354d",
      }
    }
    viewBox="0 0 800 600"
  >
    <g
      className="rsm-geographies "
      data-testid="map"
    />
  </svg>
  <div
    style={
      Object {
        "margin": "16px 0",
        "textAlign": "center",
      }
    }
  >
    <h4>
      United States
      , 
      US
    </h4>
    <h6>
      Date: 
      1/19/1970
        
      <span>
        Air Quality Index: 
      </span>
      100
    </h6>
  </div>
  <table
    style={
      Object {
        "backgroundColor": "#da4c7a",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <thead>
      <tr
        className="fs-4"
      >
        <th
          className="py-2"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          Gases
        </th>
        <th
          className="py-2"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          Pollutant concentration in μg/m3
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          PM25
        </td>
        <td
          className="py-3 fs-5"
        >
          12
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          PM10
        </td>
        <td
          className="py-3 fs-5"
        >
          20
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          NO2
        </td>
        <td
          className="py-3 fs-5"
        >
          30
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          SO2
        </td>
        <td
          className="py-3 fs-5"
        >
          40
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          CO
        </td>
        <td
          className="py-3 fs-5"
        >
          50
        </td>
      </tr>
      <tr
        className="table-row"
      >
        <td
          className="py-3 fs-5"
        >
          O3
        </td>
        <td
          className="py-3 fs-5"
        >
          60
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders correctly 1`] = `null`;
